pylightgbm/models.py

```diff
@@ -89,7 +90,10 @@ def fit(self, X, y, test_data=None):
             # Extracting best round from raw logs: 'best iteration round is'
             pattern = re.compile("best iteration round is ((\d+))")
             match = re.search(pattern, process)
-            self.best_round = int(match.group(1))
+            if match:
+                self.best_round = int(match.group(1))
+            else:
```

Comment:
This will mislead the user when there is no `best_round`. A `raise` condition would be appreciated here

The condition:
`if test_data and self.param['early_stopping_round'] > 0:`
is supposed to enforce lightgbm to, somehow, provide the best iteration in the logs.